app/src/main/java/com/gpayinterns/chat/ContactsRecyclerAdapter.java

```diff
@@ -125,16 +147,69 @@ public Filter getFilter()
-                    for(User u : mUsers)
-                    {
-                        String userName = u.username;
-                        if(userName.toLowerCase().contains(constraint.toString()))
+                    String URL = BASE_URL + GET_USERS_MOBILE
+                            +"/" + mCurrentUser + "/?" + MOBILE_PREFIX + "=" + constraint.toString();
+                    Log.d("URL",URL);
+                    JsonObjectRequest jsonObjectRequest = new JsonObjectRequest
+                            (Request.Method.GET, URL, null, new Response.Listener<JSONObject>()
+                            {
+                                @Override
+                                public void onResponse(JSONObject response)
+                                {
+                                    free[0] = true;
+                                    Log.d("ResponseMessage: " , response.toString());
+                                    try
+                                    {
+                                        JSONArray chats = response.getJSONArray("Users");
+                                        for(int i=0;i<chats.length();i++)
+                                        {
+                                            JSONObject chat = (JSONObject) chats.get(i);
+
+                                            String username = chat.getString("Username");
+                                            filteredResult.add(new User(username,null,null));
+                                        }
+                                    }
+                                    catch (JSONException e)
+                                    {
+                                        e.printStackTrace();
+                                    }
+                                }
+                            }, new Response.ErrorListener()
+                            {
+                                @Override
+                                public void onErrorResponse(VolleyError error)
+                                {
+                                    free[0] = true;
+                                }
+                            }){
+                        @Override
+                        public Priority getPriority()
                         {
-                            filteredResult.add(u);
+                            return Priority.IMMEDIATE;
                         }
+                    };
+
+                    VolleyController.getInstance(mContext).addToRequestQueueWithRetry(jsonObjectRequest);
+                }
+                else
+                {
+                    free[0]=true;
+                    filteredResult.addAll(mUsers);
+                }
+                long entryTime = System.currentTimeMillis();
+                while(!free[0] && System.currentTimeMillis()-entryTime<=9000)//wait for maximum 9 seconds
+                {
+                    try
+                    {
+                        Thread.sleep(200);
```

Comment:
What is this for?
Please add a comment.

Reply:
I've added a comment describing the reason behind this.

There must be standard libraries to better perform this task.
Should I explore those?

Reply:
Ah, I’ve just found something known as 
CountdownLatch.
I’ll be replacing those